lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -161,6 +169,12 @@ internal sealed class IMRUDriver<TMapInput, TMapOutput, TResult, TPartitionType>
             _serviceAndContextConfigurationProvider =
                 new ServiceAndContextConfigurationProvider<TMapInput, TMapOutput, TPartitionType>(dataSet, configurationManager);
 
+            if (lifecycleManager != null)
+            {
+                var handle = lifecycleManager.Subscribe(this as IObserver<IJobCancelled>);
+                _disposableResources.Add(handle);
```

Comment:
changed the logic to stop timer when all observers are gone. For symmetry also ensured that timer starts only when on first subscriber . This removes dependency on DriverStarted event (removed corresponding code)